test/base/array.jl

```diff
@@ -55,9 +55,13 @@ mutable struct MyBadType
     a::Any
 end
 const MyBadType2 = Union{BigFloat, Float32}
+struct MyBadType3
```

Comment:
cc @vchuravy; I was surprised this is `allocatedinline` even though the field itself isn't.

Reply:
I'll just note here that we discussed this on Slack - the reason this happens is the *struct itself* is allocated inline, but its member fields may not be.